README.md

```diff
@@ -67,3 +67,65 @@ jobs:
           chmod +x dist/bin/*
           dist/bin/${{ matrix.bin }} --version
 ```
+
+### [`cabal-ghc-compat`](./actions/cabal-ghc-compat/action.yml)
+
+Check that the Cabal configuration file is compatible with the version of the
+Cabal library bundled with GHC.
+
+By default, commands like `cabal build` will process the Cabal configuration
+files (`.cabal`, `cabal.project{,.freeze}`) using the version of the Cabal
+library that is bundled with the `cabal-install` tool. However, if a project
+supports older versions of GHC, it can be helpful to check that the Cabal
```

Comment:
Paging @kquick: Can you help me out with the motivation here? I wasn't sure what to say, so I just said "it can be helpful". It might be nice to specifically outline what can go wrong when one *doesn't* perform this check.

Reply:
If you try to parse the `.cabal` file with an older version of GHC+cabal library, it could fail if you are using elements in your cabal file that aren't implemented in that older cabal.  I guess you could add "if this check isn't done, the version of cabal you are using may be newer than the cabal available to the user and they may be unable to parse the cabal file."  Do you have a rewording to suggest that would conveay the motivation better?

Reply:
> If you try to parse the .cabal file with an older version of GHC+cabal library, it could fail if you are using elements in your cabal file that aren't implemented in that older cabal.

Right, understood. I'm still not clear on who would be in this situation, and how it would arise for them. Is the idea that you (or your users) might be getting your `cabal-install` from an OS package manager, which might have out-of-date versions of both?

And can you contrast the check performed by this action with e.g., specifying multiple versions of Cabal to your `setup-haskell` action via a `matrix`?

Reply:
Yes, the user may not have install rights on their system, and may be using an older OS revision that only has an older GHC and cabal available.

I don't see a `setup-haskell` action here?

Reply:
Ah, I was just typing out a more specific question :smile: Can you contrast the use of this action with using a `matrix` like so?
```yaml
strategy:
  matrix:
    # NOTE! Each GHC version should use the version of Cabal that is bundled with it, see
    # https://gitlab.haskell.org/ghc/ghc/-/wikis/commentary/libraries/version-history
    include:
    - ghc: 9.12.1
      cabal: 3.14.1.0
    - ghc: 9.10.1
      cabal: 3.12.0.0
```

Reply:
Thank you for the details :pray: I'll add a bit more color to the README!

Reply:
I've significantly expanded this part of the README, incorporating your thoughts above. Let me know how it looks!